includes/core-changes/serialization/3.1/jsonserializer-deserialize-throws-jsonexception.md

```diff
@@ -0,0 +1,35 @@
+### JsonSerializer.Deserialize throws JsonException for null or empty strings
+
+When the type parameter is <xref:System.Char>, <xref:System.Text.Json.JsonSerializer.Deserialize%60%601(System.String,System.Text.Json.JsonSerializerOptions)?displayProperty=nameWithType> now throws a <xref:System.Text.Json.JsonException> instead of a <xref:System.NullReferenceException> or <xref:System.IndexOutOfRangeException>.
+
+#### Change description
+
+In previous .NET versions, when the type parameter is <xref:System.Char>, passing a null reference for the <xref:System.String> parameter of <xref:System.Text.Json.JsonSerializer.Deserialize%60%601(System.String,System.Text.Json.JsonSerializerOptions)?displayProperty=nameWithType> causes a <xref:System.NullReferenceException> to be thrown. Passing an empty string causes an <xref:System.IndexOutOfRangeException> to be thrown.
+
+In .NET Core 3.1 and later, when the type parameter is <xref:System.Char>, passing a null reference or an empty string causes a <xref:System.Text.Json.JsonException> to be thrown.
```

Comment:
NIT: I think adding the example code might make it more clear, up to you
```csharp
    // Before (3.0): throws NullReferenceException
    // After (3.1): throw JsonException
    JsonSerializer.Deserialize<char>("null");

    // Before (3.0): throws IndexOutOfRangeException
    // After (3.1): throw JsonException
    JsonSerializer.Deserialize<char>("");

Reply:
Okay, I will add it. But isn't `"null"` just a string that contains the characters n, u, l, and l?

Reply:
Good question, it might supposed to be `JsonSerializer.Deserialize<char>(null);`, but also "null" string might be serialized into `null` value instead of "null" string cc @layomia @ahsonkhan

Reply:
I responded to this question below:
https://github.com/dotnet/docs/pull/20956#discussion_r500707091

This is invalid:
```C#
JsonSerializer.Deserialize<char>(null);
```

The example needs to be a null string in quotes, i.e. the six characters ", n, u, l, l, ".